src/main/java/seedu/taskify/logic/commands/TagSearchCommand.java

```diff
@@ -37,10 +37,15 @@ public CommandResult execute(Model model) {
             model.updateFilteredTaskList(predicate);
             return new CommandResult(
                     String.format(Messages.MESSAGE_TASKS_LISTED_OVERVIEW, model.getFilteredTaskList().size()));
-        } else {
+        } else if (CommandResult.isCompletedTab()) {
             model.updateCompletedFilterTaskList(predicate);
             return new CommandResult(
                     String.format(Messages.MESSAGE_TASKS_LISTED_OVERVIEW, model.getCompletedFilteredTaskList().size()));
+        } else {
+            model.updateUncompletedFilterTaskList(predicate);
+            return new CommandResult(
+                    String.format(Messages.MESSAGE_TASKS_LISTED_OVERVIEW,
+                            model.getUncompletedFilteredTaskList().size()));
         }
 
```

Comment:
I think it would be better code quality wise to add an `else if` clause for `CommandResult.isUncompletedTab`, then the else clause throws an assert false